File: p1/RWNums.java
import java.io.*;

public class RWNums {

    static int readSplit(InputStream is, char[] buff, char s, int maxlen) throws IOException{
        for(int i=0; i<maxlen-1; i++)
        {
          int c = is.read();
          if(c==-1)
          {
            buff[i+1] = '\0';
            return -1;
          }
          if(((char)c)==s)
          {
            buff[i+1] = '\0';
            return 0;
          }
          buff[i] = (char)c;
        }
        buff[maxlen-1] = '\0';
        return -1;
    }
    
    public static void main(String[] args) {
      int res = 0;
      char[] buff = new char [100];
      Float num;
      
      if(args.length < 3) {
         System.out.println("Usage: java RWFile infile outfile num");
         return;
      }
      num = Float.parseFloat(args[2]);
        
      try {
        InputStream is = new FileInputStream(args[0]);
        OutputStream os = new FileOutputStream(args[1]);
         
         while(res != -1) {
             res = readSplit(is,buff,' ', 100);
             if(res != -1) {
                Float f = num *Float.parseFloat(new String(buff));
                String out = f.toString();
                os.write(out.getBytes());
                os.write(' ');
             }
         }
         is.close();
         os.close();
      } catch (Exception ex) {  ex.printStackTrace(); }        
    }
}

File: p1/RWNums.py
#!/usr/bin/python3
import sys

if len(sys.argv)!=4:
	print("Usage: %s infile outfile num"%sys.argv[0])
	sys.exit(0)

infile = open(sys.argv[1], "r")
outfile = open(sys.argv[2], "w")
num = float(sys.argv[3])

cont = True
while cont:
    curnum = ""
    c = infile.read(1)
    while c!=" " and c!="":
        curnum+=c
        c = infile.read(1)
    if c=="":
        cont = False
    try:
        outnum = num*float(curnum)
        outfile.write("%f "%outnum)
    except:
        pass

infile.close()
outfile.close()
